cin_validator/rules/cin2022_23/rule_8740.py

```diff
@@ -0,0 +1,205 @@
+                "S47ActualStartDate": "26/05/1999",
+                "DateOfInitialCPC": pd.NA,
+                "ICPCnotRequired": "0",
+            },
+            {  # 6 - Ignore
+                "LAchildID": "child1",
+                "S47ActualStartDate": "26/05/1999",
+                "DateOfInitialCPC": pd.NA,
+                "ICPCnotRequired": "1",
+            },
+        ]
+    )
+    # if rule requires columns containing date values, convert those columns to datetime objects first. Do it here in the test_validate function, not above.
+    section_47s[S47ActualStartDate] = pd.to_datetime(
+        section_47s[S47ActualStartDate], format="%d/%m/%Y", errors="coerce"
+    )
+    section_47s[DateOfInitialCPC] = pd.to_datetime(
+        section_47s[DateOfInitialCPC], format="%d/%m/%Y", errors="coerce"
+    )
+
+    # Run rule function passing in our sample data
+    result = run_rule(validate, {Section47: section_47s, Header: fake_header})
+
+    # Use .type1_issues to check for the result of .push_type1_issues() which you used above.
+    issues = result.type1_issues
+
+    # get table name and check it. Replace ChildProtectionPlans with the name of your table.
+    issue_table = issues.table
+    assert issue_table == Section47
+
+    # check that the right columns were returned. Replace CPPstartDate and CPPendDate with a list of your columns.
+    issue_columns = issues.columns
+    assert issue_columns == [S47ActualStartDate, DateOfInitialCPC, ICPCnotRequired]
+
+    # check that the location linking dataframe was formed properly.
+    issue_rows = issues.row_df
+    # replace 2 with the number of failing points you expect from the sample data.
+    assert len(issue_rows) == 2
+    # check that the failing locations are contained in a DataFrame having the appropriate columns. These lines do not change.
+    assert isinstance(issue_rows, pd.DataFrame)
+    assert issue_rows.columns.to_list() == ["ERROR_ID", "ROW_ID"]
+
+    # Create the dataframe which you expect, based on the fake data you created. It should have two columns.
+    # - The first column is ERROR_ID which contains the unique combination that identifies each error instance, which you decided on earlier.
+    # - The second column in ROW_ID which contains a list of index positions that belong to each error instance.
+
+    # The ROW ID values represent the index positions where you expect the sample data to fail the validation check.
+    expected_df = pd.DataFrame(
+        [
+            {
+                "ERROR_ID": (
+                    "child1",
+                    pd.to_datetime("26/05/1999", format="%d/%m/%Y", errors="coerce"),
+                ),
+                "ROW_ID": [4],
+            },
+            {
+                "ERROR_ID": (
+                    "child1",
+                    pd.to_datetime("26/05/1999", format="%d/%m/%Y", errors="coerce"),
+                ),
+                "ROW_ID": [5],
+            },
+        ]
```

Comment:
The ROW_IDs are grouped according to their ERROR_IDs so this will be:

> {
"ERROR_ID": (
"child1",
pd.to_datetime("26/05/1999", format="%d/%m/%Y", errors="coerce"),
),
"ROW_ID": [4,5],
},